crates/astria-sequencer-relayer/src/data_availability.rs

```diff
@@ -355,128 +305,143 @@ impl CelestiaClient {
             .blob_get_all(req)
             .await
             .wrap_err("failed getting namespaced data")?;
-
-        // retrieve all sequencer data stored at this height
-        // optionally, only find data that was signed by the given public key
-        // NOTE: there should NOT be multiple datas with the same block hash and signer;
-        // should we check here, or should the caller check?
         let sequencer_namespace_datas = rsp
             .blobs
             .into_iter()
             .filter_map(|blob| {
-                let data =
-                    SignedNamespaceData::<SequencerNamespaceData>::from_bytes(&blob.data).ok()?;
-                let Some(verification_key) = verification_key else {
-                    return Some(data);
-                };
-
-                let hash = data.data.hash().ok()?;
-                let signature = Signature::try_from(&*data.signature.0).ok()?;
-                verification_key.verify(&signature, &hash).ok()?;
-                Some(data)
+                match SignedNamespaceData::<SequencerNamespaceData>::from_bytes(&blob.data) {
+                    Ok(data) => Some(data),
+                    Err(e) => {
+                        warn!(error.msg = %e, error.cause_chain = ?e, "failed deserializing sequencer namespace data from bytes stored in retrieved celestia blob");
+                        None
+                    }
+                }
             })
             .collect::<Vec<_>>();
         Ok(sequencer_namespace_datas)
     }
 
-    /// get_sequencer_block returns the full SequencerBlock (with all rollup data) for the given
-    /// SequencerNamespaceData.
-    pub async fn get_sequencer_block(
+    /// get_sequencer_block returns the full SequencerBlock (with all rollup data) for the
+    pub async fn get_all_rollup_data_from_sequencer_namespace_data(
```

Comment:
Am I correct in my reading that if there are N blocks per Celestia block, this function will be called N times?Which will result in many calls to celestia to retrieve the same data and filter it out? The number of calls to celestia grows by a factor of O(N)?

Can we make this be O(1) by not filtering out and throwing away the data we will later refetch and process?

Reply:
I think this is also out of scope for this, like makes sense to fix alongside grabbing only local namespaced data.

Reply:
does reading from celestia have any cost? i was under the impression it didn't (other than processing time ofc)

Reply:
ideally this will only be called once, passing in the `namespace_data` which has already been verified as being the correct data. also this should probably be moved to conductor (or some shared crate if this is used in relayer tests?)

Reply:
This here exists only for conductor I think. It should be refactored and moved out of the relayer so that conductor doesn't import the whole thing.

Re cost: you mean in fees?

Reply:
https://github.com/astriaorg/astria/issues/175